contracts/artifacts/DelegatedOptIn.approval.teal

```diff
@@ -0,0 +1,165 @@
+#pragma version 9
+
+txn ApplicationID
+int 0
+>
+int 6
+*
+txn OnCompletion
++
+switch create_NoOp NOT_IMPLEMENTED NOT_IMPLEMENTED NOT_IMPLEMENTED NOT_IMPLEMENTED NOT_IMPLEMENTED call_NoOp
+
+NOT_IMPLEMENTED:
+	err
+
+abi_route_setSignature:
+	byte 0x
+	txn GroupIndex
+	int 1
+	-
+	txna ApplicationArgs 1
+	callsub setSignature
+	int 1
+	return
+
+setSignature:
+	proto 3 0
+
+	// Record MBR before box_put to later determine the MBR delta
+	// ./contracts/delegated_optin_app.algo.ts:22
+	// preMBR = this.app.address.minBalance
+	txna Applications 0
+	app_params_get AppAddress
+	assert
+	acct_params_get AcctMinBalance
+	assert
+	frame_bury -3 // preMBR: uint64
+
+	// ./contracts/delegated_optin_app.algo.ts:23
+	// this.signatures.set(this.txn.sender, sig)
+	txn Sender
+	frame_dig -1 // sig: byte[64]
+	box_put
+
+	// Verify box MBR payment
+	// ./contracts/delegated_optin_app.algo.ts:26
+	// assert(boxMBRPayment.receiver === this.app.address)
+	frame_dig -2 // boxMBRPayment: pay
+	gtxns Receiver
+	txna Applications 0
+	app_params_get AppAddress
+	assert
+	==
+	assert
+
+	// ./contracts/delegated_optin_app.algo.ts:27
+	// assert(boxMBRPayment.amount >= this.app.address.minBalance - preMBR)
+	frame_dig -2 // boxMBRPayment: pay
+	gtxns Amount
+	txna Applications 0
+	app_params_get AppAddress
+	assert
+	acct_params_get AcctMinBalance
+	assert
+	frame_dig -3 // preMBR: uint64
+	-
+	>=
+	assert
+	retsub
+
+abi_route_delegatedOptIn:
+	// no dupn needed
```

Comment:
What does this comment mean? And why wasn't it emitted at the start of the other `abi_` method, which also did not begin with `dupn`?

Reply:
TEALScript uses the frame for local variables, so it needs to push values to the stack when there are local variables prior to calling the subroutine. `// no dupn needed` is basically saying that there doesn't need to any extra values pushed to the stack since there are no local variables. 

This comment will be improved along with more comments for clarity in a near-future update

Reply:
Resolved in c5d72037dcc7c69de47a1e8a018707090a949b9e